qiskit_experiments/library/characterization/t1_analysis.py

```diff
@@ -12,202 +12,40 @@
+        extra = self._get_option("extra")
 
-        xdata, ydata, sigma = process_curve_data(data, lambda datum: level2_probability(datum, "1"))
-        xdata *= conversion_factor
-
-        if t1_guess is None:
-            t1_guess = np.mean(xdata)
-        else:
-            t1_guess = t1_guess * conversion_factor
-        if offset_guess is None:
-            offset_guess = ydata[-1]
-        if amplitude_guess is None:
-            amplitude_guess = ydata[0] - offset_guess
-
-        # Perform fit
-        def fit_fun(x, a, tau, c):
-            return a * np.exp(-x / tau) + c
-
-        init = {"a": amplitude_guess, "tau": t1_guess, "c": offset_guess}
-        fit_result = curve_fit(fit_fun, xdata, ydata, init, sigma=sigma)
-        fit_result = dataclasses.asdict(fit_result)
-        fit_result["circuit_unit"] = unit
-        if unit == "dt":
-            fit_result["dt"] = conversion_factor
-
-        # Construct analysis result
-        name = "T1"
-        unit = "s"
-        value = FitVal(fit_result["popt"][1], fit_result["popt_err"][1], unit="s")
-        chisq = fit_result["reduced_chisq"]
-        quality = self._fit_quality(
-            fit_result["popt"], fit_result["popt_err"], fit_result["reduced_chisq"]
-        )
-        analysis_results = [
-            AnalysisResultData(
-                name,
-                value,
-                chisq=chisq,
-                quality=quality,
-                extra=fit_result,
-            )
-        ]
-
-        # Generate fit plot
-        figures = []
-        if plot:
-            ax = plot_curve_fit(fit_fun, fit_result, ax=ax, fit_uncertainty=True)
-            ax = plot_errorbar(xdata, ydata, sigma, ax=ax)
-            self._format_plot(ax, fit_result, qubit=qubit)
-            figures.append(ax.get_figure())
-
-        return analysis_results, figures
-
-    @staticmethod
-    def _fit_quality(fit_out, fit_err, reduced_chisq):
-        # pylint: disable = too-many-boolean-expressions
-        if (
-            abs(fit_out[0] - 1.0) < 0.1
-            and abs(fit_out[2]) < 0.1
-            and reduced_chisq < 3
-            and (fit_err[0] is None or fit_err[0] < 0.1)
-            and (fit_err[1] is None or fit_err[1] < fit_out[1])
-            and (fit_err[2] is None or fit_err[2] < 0.1)
-        ):
-            return "good"
-        else:
-            return "bad"
-
-    @classmethod
-    def _format_plot(cls, ax, analysis_result, qubit=None, add_label=True):
-        """Format curve fit plot"""
-        # Formatting
-        ax.tick_params(labelsize=14)
-        if qubit is not None:
-            ax.set_title(f"Qubit {qubit}", fontsize=16)
-        ax.set_xlabel("Delay (s)", fontsize=16)
-        ax.ticklabel_format(axis="x", style="sci", scilimits=(-3, 3))
-        ax.set_ylabel("P(1)", fontsize=16)
-        ax.grid(True)
+        conversion_factor = extra.get("conversion_factor", 1)
+        user_opt.p0["tau"] *= conversion_factor
```

Comment:
This is really unique to this experiment. For example, spectroscopy experiment can take arbitrary unit but doesn't manage prefactor like this. Usually users don't need to manually specify the init guess if algorithmic guesses are enough smart. So removing this logic doesn't hurt usability and saves us from maintenance & extra unittest for this.

Reply:
Will be done in follow-up PR